WalletWasabi/Nito/AsyncEx/AsyncMutex.cs

```diff
@@ -379,6 +397,38 @@ private void ReleaseLock()
 			ChangeStatus(AsyncLockStatus.StatusReady, AsyncLockStatus.StatusReleasing);
 		}
 
+		public static async Task WaitForAllMutexToCloseAsync()
+		{
+			DateTime start = DateTime.Now;
+			lock (AsyncMutexesLock)
+			{
+				foreach (var mutex in AsyncMutexes)
+				{
+					mutex.Value.IsQuitPending = true;
+				}
+			}
+
+			while (true)
+			{
+				lock (AsyncMutexesLock)
+				{
+					bool stillRunning = AsyncMutexes.Where(am => am.Value.IsAlive).Any();
+					if (!stillRunning)
+					{
+						return;
+					}
+					Logger.LogDebug($"Waiting for: {string.Join(", ", AsyncMutexes.Where(am => am.Value.IsAlive).Select(m => m.Value.ShortName))}", nameof(AsyncMutex));
+				}
+				await Task.Delay(200);
```

Comment:
I've been playing with this method (not on this branch.) And it seems to only work if I use the exact same hack here as everywhere else below: `Task.Delay(200).GetAwaiter().GetResult();` I didn't try this branch.